cocos/core/pipeline/pipeline-ubo.ts

```diff
@@ -511,7 +511,7 @@ export class PipelineUBO {
         globalDS.bindTexture(UNIFORM_SHADOWMAP_BINDING, builtinResMgr.get<Texture2D>('default-texture').getGFXTexture()!);
         globalDS.bindTexture(UNIFORM_SPOT_SHADOW_MAP_TEXTURE_BINDING, builtinResMgr.get<Texture2D>('default-texture').getGFXTexture()!);
         globalDS.update();
-        globalDS.getBuffer(UBOShadow.BINDING).update(this._shadowUBO);
+        this._pipeline.commandBuffers[0].updateBuffer(globalDS.getBuffer(UBOShadow.BINDING), this._shadowUBO);
```

Comment:
Why change it like this?

Reply:
It's a bug, it should be updated by commandbuffer

Reply:
Can any test case trigger the error?

Reply:
Nope, this works well with webGL, but webGL seems delayed every update until it has to. Same implementation in c++ doing this way.

Reply:
Ok.